### PR TITLE
[ACS-6231] Nx Workspace 17 and Node 20

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**
```
[x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-js-api/wiki/Commit-format)
[ ] Tests for the changes have been added (for bug fixes / features)
[ ] Docs have been added / updated (for bug fixes / features)
```

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Documentation
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

- migrate project to Nx 17
- switch to node 20
- use the es2022 for builds
- setup mocha for testing
- cleanup unused or obsolete files
- add missing report api types and models

image

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://alfresco.atlassian.net/browse/ACS-6231

### CODE REVIEW

File: test/content-services/tagApi.spec.ts [hunk 1]
```diff
@@ -67,37 +67,41 @@ describe('Tags', () => {
             );
         });
 
-        it('should return specified tag', (done) => {
+        it.skip('should return specified tag', (done) => {
```
Comment (AleksanderSklorz) [inline test/content-services/tagApi.spec.ts:70]: Do we need that skip?

File: test/content-services/tagApi.spec.ts [hunk 2]
```diff
@@ -67,37 +67,41 @@ describe('Tags', () => {
...
+                        id: '0d89aa82-f2b8-4a37-9a54-f4c5148174d6'
+                    });
+                    done();
                 });
-                done();
-            });
         });
 
-        it('should return tags contained specified value', (done) => {
+        it.skip('should return tags contained specified value', (done) => {
```
Comment (AleksanderSklorz) [inline test/content-services/tagApi.spec.ts:86]: Do we need that skip?

Comment (AleksanderSklorz) [reply test/content-services/tagApi.spec.ts:86]: Will be that done as part of that PR or could you please raise jira for this?

Comment (DenysVuika) [reply test/content-services/tagApi.spec.ts:86]: I am investigating, no idea why 2 tests are failing so far

Comment (DenysVuika) [reply test/content-services/tagApi.spec.ts]: fixed

File: test/content-services/tagApi.spec.ts (hunk 1 above)
Comment (DenysVuika) [reply test/content-services/tagApi.spec.ts]: fixed

**Review by AleksanderSklorz — APPROVED** [review]

**Pull request merged**